### PR TITLE
Reputation changes

### PR DESCRIPTION
**Pull request by HitoriWoB**
Added double reputation event
Added +1 static FS reputation

### CODE REVIEW

**Comment by aumu1031** [comment]
Can you make this also apply the multiplier when getting caps?

**Comment by aumu1031** [comment]
You can reuse the same Event display page by copy/pasting this on the `temp_event.php` page:

```<?php elseif ($system->event instanceof DoubleExpEvent): ?>
<table class="table">
    <tr>
        <th>Event Details</th>
    </tr>
    <tr style="text-align: center">
        <td>Double Experience is active until <?= $system->event->end_time->format("l, M jS, Y")?>.</td>
    </tr>
</table>
<?php else: ?>
    <table class="table">
        <tr><th>Event Details</th></tr>
        <tr><td>There is no active event.</td></tr>
    </table>
<?php endif; ?>```

File: classes/ForbiddenSeal.php [hunk 1]
```diff
@@ -32,8 +32,8 @@ class ForbiddenSeal {
     public float $extended_training_time;
     public float $extended_training_gains;
     public int $free_transfer_bonus;
-
     public int $max_battle_history_view;
+    public int $bonus_reputation;
```
Comment (levimeahan) [inline classes/ForbiddenSeal.php:36]: Let's call this out as `bonus_pve_reputation` to be clear

File: classes/event/DoubleReputation.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+require_once __DIR__ . '/Event.php';
+
+class DoubleReputation extends Event
```
Comment (levimeahan) [inline classes/event/DoubleReputation.php:5]:
```suggestion
class DoubleReputationEvent extends Event
```
Following convention here

File: classes/UserReputation.php [hunk 3]
```diff
@@ -247,6 +255,14 @@ public function __construct(&$player_rep, &$player_weekly_rep, &$player_pvp_rep,
      * Returns amount of reputation awarded for display/data confirmation purposes
      */
     public function addRep(int $amount, bool $bypass_weekly_cap = false, bool $increment_pvp = false): int {
+        // Double repuation
+        if (!empty($this->event) && $this->event instanceof DoubleReputation) {
+            $amount *= DoubleReputation::rep_modifier;
+        }
+
+        //Bonus seal reputation, 0 by default no need to check, bonus rep amounts are set by seal loading
+        $amount += $this->bonus_rep;
```
Comment (levimeahan) [inline classes/UserReputation.php:264]: This seems like it would apply to PvP too, right? Per last night I think we want to exclude PvP from these gains

**Review by levimeahan — COMMENTED** [review]
Most of this looks good, but we need the double rep event to double rep caps as well

File: classes/UserReputation.php (hunk 3 above)
Comment (HitoriWoB) [reply classes/UserReputation.php:264]: That's correct. Just hadn't had a chance to factor it in.

File: classes/User.php [hunk 4]
```diff
@@ -951,7 +951,7 @@ public function loadTrainingManager(): void
             $this->bloodline_id);
     }
 
-    public function setForbiddenSealFromDb(string $forbidden_seal_db, bool $remote_view) {
+    public function setForbiddenSealFromDb(string $forbidden_seal_db, bool $remote_view, bool $set_seal_bonus) {
```
Comment (levimeahan) [inline classes/User.php:954]: An internal flag might be better on this - E.g, the first time we call this, set the bonuses and set `$this->seal_bonuses_applied = true`, the second time, we check the `seal_bonuses_applied` flag and skip applying them (or override the old ones)

File: classes/event/DoubleReputationEvent.php [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+require_once __DIR__ . '/Event.php';
+
+class DoubleReputationEvent extends Event
+{
+    const rep_modifier = 2;
+    const pve_modifier = 2;
+    const pvp_modifier = 1;
```
Comment (levimeahan) [inline classes/event/DoubleReputationEvent.php:9]:
```suggestion
    const rep_gain_multiplier = 2;
    const pve_cap_multiplier = 2;
    const pvp_cap_multiplier = 2;
```

File: classes/UserReputation.php [hunk 6]
```diff
@@ -247,8 +261,16 @@ public function __construct(&$player_rep, &$player_weekly_rep, &$player_pvp_rep,
      * Returns amount of reputation awarded for display/data confirmation purposes
      */
     public function addRep(int $amount, bool $bypass_weekly_cap = false, bool $increment_pvp = false): int {
+        // Double repuation
+        if (!empty($this->event) && $this->event instanceof DoubleReputationEvent) {
+            $amount *= DoubleReputationEvent::rep_modifier;
```
Comment (levimeahan) [inline classes/UserReputation.php:266]:
```suggestion
            $amount *= DoubleReputationEvent::rep_gain_multiplier;
```

File: classes/UserReputation.php [hunk 7]
```diff
@@ -232,6 +240,12 @@ public function __construct(&$player_rep, &$player_weekly_rep, &$player_pvp_rep,
         $this->weekly_cap = $REP_RANK['weekly_cap'];
         $this->weekly_pvp_cap = $REP_RANK['weekly_pvp_cap'];
         $this->base_pvp_reward = $REP_RANK['base_pvp_rep_reward'];
+
+        // EVENT MODIFICATIONS
+        if(!empty($this->event) && $this->event instanceof DoubleReputationEvent) {
+            $this->weekly_cap *= DoubleReputationEvent::pve_modifier;
+            $this->weekly_pvp_cap *= DoubleReputationEvent::pvp_modifier;
```
Comment (levimeahan) [inline classes/UserReputation.php:247]:
```suggestion
            $this->weekly_cap *= DoubleReputationEvent::pve_cap_multiplier;
            $this->weekly_pvp_cap *= DoubleReputationEvent::pvp_cap_multiplier;
```

**Review by levimeahan — APPROVED** [review]

File: classes/event/DoubleReputationEvent.php [hunk 8]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+require_once __DIR__ . '/Event.php';
+
+class DoubleReputationEvent extends Event
+{
+    const rep_gain_multiplier = 2;
+    const pve_cap_multiplier = 2;
+    const pvp_cap_multiplier = 1;
```
Comment (levimeahan) [inline classes/event/DoubleReputationEvent.php:9]:
```suggestion
    const pvp_cap_multiplier = 2;
```

**Review by levimeahan — APPROVED** [review]

**Pull request merged**